bokeh/client/session.py

```diff
@@ -281,6 +306,14 @@ def _document_changed(self, event):
             log.debug("Not sending notification back to server for a change it requested")
             return
 
+        if isinstance(event, PeriodicCallbackAdded):
+            self.add_periodic_callback(event.callback)
+            return
+
+        if isinstance(event, PeriodicCallbackRemoved):
+            self.remove_periodic_callback(event.callback)
+            return
+
```

Comment:
+1

We really have 2 main aspects here.. one is `should_sync_over_websocket` and the other is the fact that `PeriodicCallbackAdded` and `PeriodicCallbackRemoved` require an action from session. 

We already addressed the first issue (well, will address is it :) ) but regarding the second (having actions to be performed by session upon an event) I was considering adding a Mixin (that both `client` and `server` sessions can use) that encapsulate these actions into a `act_on_event` method.. What do you think?